### PR TITLE
Support kubeconfigs for admission plugin configurations for `kube-apiserver`

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area usability
/kind enhancement

**What this PR does / why we need it**:
This PR allows to configure the `kubeapiserver` component with kubeconfigs which should be used as part of admission plugins. Today, only three admission plugins are supporting this.

Note that this PR does not yet expose this via the `Shoot` resource in any way. This can be done separately.
Generally, this is a prerequisite for `gardener-operator` managing the `kube-apiserver` deployment.

**Which issue(s) this PR fixes**:
Part of #2879

**Special notes for your reviewer**:
/cc @shafeeqes 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by shafeeqes — COMMENTED** [review]
Very nice, Thanks!
/lgtm

File: pkg/operation/botanist/component/kubeapiserver/configmaps.go [hunk 1]
```diff
@@ -79,6 +85,89 @@ func admissionPluginsConfigFilename(name string) string {
...
+	var (
+		nothingToMutate    = (plugin.Config == nil || plugin.Config.Raw == nil) && len(plugin.Kubeconfig) == 0
+		mustDefaultConfig  = (plugin.Config == nil || plugin.Config.Raw == nil) && len(plugin.Kubeconfig) > 0
+		kubeconfigFilePath = volumeMountPathAdmissionKubeconfigSecrets + "/" + admissionPluginsKubeconfigFilename(plugin.Name)
+	)
+
+	if len(plugin.Kubeconfig) == 0 {
+		// This makes sure that the path to the kubeconfig is overwritten if specified in case no kubeconfig was
+		// provided. It prevents that users can access arbitrary files in the kube-apiserver pods and disguise them as
+		// kubeconfigs for their admission plugin configs.
```
Comment (shafeeqes) [inline pkg/operation/botanist/component/kubeapiserver/configmaps.go:98]: Just one question, What did you mean by this comment?
How can users disguise arbitrary files as kubeconfigs?

Comment (shafeeqes) [inline pkg/operation/botanist/component/kubeapiserver/configmaps.go:98]: Just one question, how can users disguise arbitrary files as kubeconfigs for the plugins? We don't add it to the admissionkubeconfig secret if `plugin.Kubeconfig` is empty.

Comment (rfranzke) [reply pkg/operation/botanist/component/kubeapiserver/configmaps.go:98]: Users could provide admission config which already contains the `kubeConfig: /some/path/to/some/file` key. Thus, without protection, they might be able to access arbitrary files in the `kube-apiserver` pod they should not have access to.

Comment (shafeeqes) [reply pkg/operation/botanist/component/kubeapiserver/configmaps.go:98]: Ah, got it!

Comment (shafeeqes) [reply pkg/operation/botanist/component/kubeapiserver/configmaps.go:98]: So, previously this was allowed, right? To provide a random file in the `kubeConfig` field config.

Comment (rfranzke) [reply pkg/operation/botanist/component/kubeapiserver/configmaps.go:98]: I'm afraid yes

**Pull request merged**